tests/e2e/utils/evm_client.go

```diff
@@ -153,34 +149,23 @@ func (ec *EvmClient) TransferTx(
 				GasFeeCap: ec.feeCap,
 				GasTipCap: ec.priorityFee,
 				Value:     transferAmount,
-				Data:      []byte{},
 			}),
 			ec.signer,
 			senderPriv,
 		)
 		if err != nil {
-			log.Printf("failed to sign transaction: %v (key address %s)", err, sender)
-			time.Sleep(time.Second)
-			continue
+			return nil, fmt.Errorf("failed to sign transaction: %w", err)
 		}
 
 		if err := ec.ethClient.SendTransaction(ctx, signedTx); err != nil {
 			log.Printf("failed to send transaction: %v (key address %s)", err, sender)
-
-			if strings.Contains(err.Error(), precompile.ErrSenderAddressNotAllowListed.Error()) {
-				return nil, err
-			}
-
-			time.Sleep(time.Second)
-			continue
+			return nil, err
 		}
 
 		txHash := signedTx.Hash()
 		cost, err := ec.ConfirmTx(ctx, txHash)
 		if err != nil {
-			log.Printf("failed to confirm %s: %v", txHash.Hex(), err)
-			time.Sleep(time.Second)
```

Comment:
ConfirmTx is already trying multiple times to confirm the transaction, no need to continue and retry the outer loop here